src/storybook/get-stories/wait-storybook-data-json.ts

```diff
@@ -0,0 +1,62 @@
+import {
+    STORYBOOK_WAIT_SERVER_TIMEOUT,
+    STORYBOOK_SERVER_REQUEST_TIMEOUT,
+    STORYBOOK_SERVER_CHECK_INTERVAL,
+} from "../../constants";
+import logger from "../../logger";
+import type { StorybookDataJson } from "./extract-stories";
+
+export const waitStorybookDataJson = async (
```

Comment:
- tries to fetch multiple sources of data (`stories.json`, `index.json`) and return the one, which we can get
- retries to fetch on request timeouts
- stop trying in given timeout
- log a warning, if it is not ECONNREFUSED (this will be thrown, while local dev server is still launching)